hs_core/tasks.py

```diff
@@ -325,41 +325,51 @@ def delete_zip(zip_path):
-def create_temp_zip(resource_id, input_path, output_path, sf_aggregation, sf_zip=False):
+def create_temp_zip(resource_id, input_path, output_path, aggregation_name=None, sf_zip=False):
     """ Create temporary zip file from input_path and store in output_path
+    :param resource_id: the short_id of a resource
     :param input_path: full irods path of input starting with federation path
     :param output_path: full irods path of output starting with federation path
-    :param sf_aggregation: if True, include logical metadata files
+    :param aggregation_name: The name of the aggregation to zip
+    :param sf_zip: signals a single file to zip
     """
     from hs_core.hydroshare.utils import get_resource_by_shortkey
     res = get_resource_by_shortkey(resource_id)
+    aggregation = None
+    if aggregation_name:
+        aggregation = res.get_aggregation_by_aggregation_name(aggregation_name)
     istorage = res.get_irods_storage()  # invoke federated storage as necessary
 
     if res.resource_type == "CompositeResource":
         if '/data/contents/' in input_path:
             short_path = input_path.split('/data/contents/')[1]  # strip /data/contents/
-            res.create_aggregation_xml_documents(aggregation_name=short_path)
+            res.create_aggregation_xml_documents(path=short_path)
         else:  # all metadata included, e.g., /data/*
             res.create_aggregation_xml_documents()
 
     try:
-        if sf_zip:
+        if aggregation or sf_zip:
             # input path points to single file aggregation
             # ensure that foo.zip contains aggregation metadata
             # by copying these into a temp subdirectory foo/foo parallel to where foo.zip is stored
             temp_folder_name, ext = os.path.splitext(output_path)  # strip zip to get scratch dir
             head, tail = os.path.split(temp_folder_name)  # tail is unqualified folder name "foo"
             out_with_folder = os.path.join(temp_folder_name, tail)  # foo/foo is subdir to zip
             istorage.copyFiles(input_path, out_with_folder)
-            if sf_aggregation:
+            if aggregation:
                 try:
-                    istorage.copyFiles(input_path + '_resmap.xml',  out_with_folder + '_resmap.xml')
+                    istorage.copyFiles(aggregation.map_file_path,  temp_folder_name)
                 except SessionException:
-                    logger.error("cannot copy {}".format(input_path + '_resmap.xml'))
+                    logger.error("cannot copy {}".format(aggregation.map_file_path))
                 try:
-                    istorage.copyFiles(input_path + '_meta.xml', out_with_folder + '_meta.xml')
+                    istorage.copyFiles(aggregation.metadata_file_path, temp_folder_name)
                 except SessionException:
-                    logger.error("cannot copy {}".format(input_path + '_meta.xml'))
+                    logger.error("cannot copy {}".format(aggregation.metadata_file_path))

```

Comment:
better to log irods SessionException error as well by retrieving it from ex.stderr